File: app/src/main/java/com/example/kingsofkode/MyAdapter.java
package com.example.kingsofkode;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

public class MyAdapter extends RecyclerView.Adapter<MyAdapter.MyViewHolder> {

    private Context context;
    private ArrayList<String> arrayList;

    public MyAdapter(Context context, ArrayList<String> arrayList) {
        this.context = context;
        this.arrayList = arrayList;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.character_list_item, parent, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        int image_id = context.getResources().getIdentifier(arrayList.get(position % arrayList.size()), "drawable", context.getPackageName());
        holder.character_imageview.setImageResource(image_id);
    }

    @Override
    public int getItemCount() {
        return Integer.MAX_VALUE;
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {
        ImageView character_imageview;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            character_imageview = itemView.findViewById(R.id.character_imageview);
        }
    }
}

File: app/src/main/java/com/example/kingsofkode/MyAdapter.kt
package com.example.kingsofkode

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import androidx.recyclerview.widget.RecyclerView.ViewHolder
import com.example.kingsofkode.MyAdapter.MyViewHolder
import java.util.*

class MyAdapter(private val context: Context, private val arrayList: ArrayList<String>) :
    RecyclerView.Adapter<MyViewHolder>() {
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view = LayoutInflater.from(context).inflate(R.layout.character_list_item, parent, false)
        return MyViewHolder(view)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val image_id = context.resources.getIdentifier(
            arrayList[position % arrayList.size],
            "drawable",
            context.packageName
        )
        holder.character_imageview.setImageResource(image_id)
    }

    override fun getItemCount(): Int {
        return Int.MAX_VALUE
    }

    inner class MyViewHolder(itemView: View) : ViewHolder(itemView) {
        var character_imageview: ImageView

        init {
            character_imageview = itemView.findViewById(R.id.character_imageview)
        }
    }
}